internal/stop/stop.go

```diff
@@ -6,12 +6,21 @@ import (
 	"fmt"
 	"io"
 
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/filters"
 	"github.com/docker/docker/api/types/volume"
 	"github.com/spf13/afero"
 	"github.com/supabase/cli/internal/utils"
 )
 
-func Run(ctx context.Context, backup bool, projectId string, fsys afero.Fs) error {
+func Run(ctx context.Context, backup bool, projectId string, all bool, fsys afero.Fs) error {
+	if all {
+		return stopAllProjects(ctx, backup, fsys)
+	}
+	return stopProject(ctx, backup, projectId, fsys)
```

Comment:
Looking at `utils.DockerRemoveAll` https://github.com/supabase/cli/blob/develop/internal/utils/docker.go#L97, it might be better to extend it with a `projectId` arg. If passed down an empty string, the `CliProjectFilter` would simply match all running projects for deletion.

Reply:
Yup, only thing is that it's not exactly the same filter. In one case, you want something like `label=supa-project=projectid` in the other you want `label=supa-project`. So I had to push the `projectId` info down to the `CliProjectFilter` where the actual docker filter is constructed.